From 36e8b08e8fcec46e455ab34acb05c297bab76599 Mon Sep 17 00:00:00 2001
From: devresse <adrien.devresse@epfl.ch>
Date: Fri, 10 Mar 2017 11:17:10 +0000
Subject: [PATCH] Decouple Brayns dependencies from git_subprojects to allow to
 build Brayns standalone  with external dependencies the option
 -DDISABLE_SUBPROJECTS=TRUE

---
 CMake/BraynsDepResolver.cmake | 55 +++++++++++++++++++++++++++++++++++++++++++
 CMakeLists.txt                |  2 ++
 2 files changed, 57 insertions(+), 35 deletions(-)
 create mode 100644 CMake/BraynsDepResolver.cmake

diff --git a/CMake/BraynsDepResolver.cmake b/CMake/BraynsDepResolver.cmake
new file mode 100644
index 0000000..ebef858
--- /dev/null
+++ b/CMake/BraynsDepResolver.cmake
@@ -0,0 +1,55 @@
+# Gather all the conditional
+# common_package dependencies of Brayns
+
+
+common_find_package(Servus REQUIRED)
+
+if(BRAYNS_NETWORKING_ENABLED OR BRAYNS_DEFLECT_ENABLED)
+  common_find_package(ZeroEQ REQUIRED)
+  common_find_package(ZeroBuf REQUIRED)
+  common_find_package(Lexis REQUIRED)
+endif()
+
+# ZeroEQ and HTTP messaging
+if(BRAYNS_NETWORKING_ENABLED)
+  common_find_package(LibJpegTurbo REQUIRED)
+endif()
+
+# Streaming to display walls
+if(BRAYNS_DEFLECT_ENABLED)
+  common_find_package(Deflect)
+endif()
+
+# Data access
+if(BRAYNS_BRION_ENABLED)
+  common_find_package(Lunchbox)
+  common_find_package(BBPTestData)
+  common_find_package(Brion REQUIRED)
+  common_find_package(HDF5 SYSTEM REQUIRED COMPONENTS C CXX)
+endif()
+
+# OSPRay rendering engine
+if(BRAYNS_OSPRAY_ENABLED)
+    common_find_package(ospray SYSTEM)
+endif()
+
+# OptiX rendering engine
+if(BRAYNS_OPTIX_ENABLED)
+    common_find_package(OptiX SYSTEM)
+    common_find_package(CUDA SYSTEM)
+endif()
+
+# Livre rendering engine
+if(BRAYNS_LIVRE_ENABLED)
+  common_find_package(Livre)
+endif()
+
+# Image processing
+if(BRAYNS_IMAGEMAGICK_ENABLED)
+  common_find_package(Magick++ SYSTEM)
+endif()
+
+# Mesh loading
+if(BRAYNS_ASSIMP_ENABLED)
+  common_find_package(assimp)
+endif()
diff --git a/CMakeLists.txt b/CMakeLists.txt
index bdaa52f..7ddb8fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,8 @@ option(BRAYNS_LIVRE_ENABLED "Activate Livre rendering engine" ON)
 
 include(Common)
 
+include(BraynsDepResolver)
+
 if(BRAYNS_OSPRAY_ENABLED)
     # ------------------------------------------------------------------------------
     # OSPRAY specific settings
-- 
1.9.1

